engines/tot/tot.h

```diff
@@ -0,0 +1,516 @@
+
+	void readConversationFile();
+
+	//Rout2
+	void adjustKey();
+	void adjustKey2();
+	void animateBat();
+	void updateVideo();
+	void nicheAnimation(byte nicheDir, int32 bitmap);
+	void replaceBackpack(byte obj1, uint obj2);
+	void dropObjectInScreen(ScreenObject replacementObject);
+	void calculateTrajectory(uint finalX, uint finalY);
+	void animatedSequence(uint numSequence);
+	void initScreenPointers();
+	void verifyCopyProtection();
+	void verifyCopyProtection2();
+
+	void loadAnimation(const Common::String &animation);
+	void loadCharAnimation();
+	void animateGive(uint dir, uint height);
+	void animatePickup1(uint dir, uint height);
+	void animatePickup2(uint dir, uint height);
+	void animateOpen2(uint dir, uint height);
+
+	void loadTV();
+	void loadScreen();
+	void loadScreenLayerWithDepth(uint coordx, uint coordy, uint bitmapSize, int32 bitmapIndex, uint depth);
+	void loadScreenLayer(uint coordx, uint coordy, uint picSize, int32 pic, uint prof);
+	void readBitmap(int32 bitmapOffset, byte *bitmap, uint bitmapSize, uint errorCode);
+	void updateAltScreen(byte altScreenNumber);
+
+	void freeInventory();
+	void updateInventory(byte index);
+	void updateObject(uint filePos);
+	void readObject(Common::SeekableReadStream *stream, uint objPos, ScreenObject &thisRegObj);
+	void saveObject(ScreenObject object, Common::SeekableWriteStream *stream);
+	void saveItemRegister();
+
+	void saveTemporaryGame();
+	void drawLookAtItem(RoomObjectListEntry obj);
+	void putIcon(uint posX, uint posY, uint iconNumber);
+	void drawInventory(byte direction, byte limit);
+	void generateDiploma(Common::String &photoName);
+	void credits();
+	void checkMouseGrid();
+	void introduction();
+	void firstIntroduction();
+	void readAlphaGraph(Common::String &data, int length, int x, int y, byte barColor);
+	void readAlphaGraphSmall(Common::String &data, int length, int x, int y, byte barColor, byte textColor);
+	void displayObjectDescription(const Common::String &text);
+	void copyProtection();
+	void initialLogo();
+	void mainMenu(bool fade);
+	void exitToDOS();
+	void soundControls();
+	void sacrificeScene();
+	void ending();
+	void loadBat();
+	void loadDevil();
+	void assembleCompleteBackground(byte *image, uint coordx, uint coordy);
+	void assembleScreen(bool scroll = false);
+	void disableSecondAnimation();
+	void clearGame();
+
+	// vars
+	void clearCurrentInventoryObject();
+	void initVars();
+	void resetGameState();
+	void clearVars();
+
+protected:
+	// Engine APIs
+	Common::Error run() override;
+
+public:
+	Graphics::Screen *_screen = nullptr;
+	Tot::GraphicsManager *_graphics = nullptr;
+	SoundManager *_sound = nullptr;
+	MouseManager *_mouse = nullptr;
+	ChronoManager *_chrono = nullptr;
+
+	bool _showMouseGrid = false;
+	bool _showScreenGrid = false;
+	bool _showGameGrid = false;
+	bool _drawObjectAreas = false;
+
+	Common::Language _lang = Common::ES_ESP;
+
+	Common::MemorySeekableReadWriteStream *_conversationData;
```

Comment:
```suggestion
	Common::MemorySeekableReadWriteStream *_conversationData = nullptr;
```
and in the rest of the pointers below in this class